docs/agents/span-min-duration.md

```diff
@@ -0,0 +1,118 @@
+# Discarding fast spans
+
+Currently,
+the experience of our APM solution is not ideal when monitoring an application that ends up generating thousands of spans per request.
+
+We currently have a default limit of 500 spans in the agents and 1000 spans in the UI to prevent overwhelming the system with too many spans.
+However,
+that just cuts off at a certain point without considering which spans are important and which are not.
+
+## Configuration option
+Key: `span_min_duration`
+
+Default value: 0ms (don't discard any spans)
+
+Description:
+Sets the minimum duration of spans.
+Spans that execute faster than this threshold are attempted to be discarded.
+
+The attempt fails if they lead up to a span that can't be discarded.
+Spans that propagate the trace context to downstream services,
+such as outgoing HTTP requests,
+can't be discarded.
+Additionally, spans that lead to an error or that may be a parent of an async operation can't be discarded.
+
+However, external calls that don't propagate context,
+such as calls to a database, can be discarded using this threshold.
+
+
+## Limitations
+
+### Spans that propagate context to downstream services can't be discarded
+We only know whether to discard after the call has ended.
+At that point,
+the trace has already continued on the downstream service.
+Discarding the span for the external request would orphan the transaction of the downstream call.
+
+An argument could be made that this is not a big problem as the trace view then just won't show the downstream transaction.
+But as this would introduce inconsistencies (e.g.
+the transaction can be seen in the transaction details view of the service but when viewing the full trace it disappears) I suggest not allowing this for now.
+
+### Intermediate spans
+Discarding a single span that has both a parent and a child span is not possible as it would lead to orphaned child spans.
+
+However,
+a whole subtree of spans can get discarded if all spans within that tree are requesting to be discarded.
+This means that if a leaf of the tree can't be discarded because it propagates context downstream,
+all spans leading up to it can't be discarded.
+If the leaf is a non-context propagating span,
+such as a manually created span or a SQL call,
+the subtree can be discarded.
+
+### Async spans
+
+If the context of a span is propagated to another thread,
+it may not be discarded.
+That is because the other thread might create child spans of the first span even if it has already ended.
+
+## Implementation
+
+Spans store two flags in order to determine whether a span can be discarded:
+- `discardable`:
+  Whether discarding this span is allowed.
+  The default value is `true`.
+  Setting this to `false` also sets the `discardable` flag of all it's parents and grand-parents to `true.
+- `discardRequested`:
+  Whether this span is 
```

Comment:
looks like part of the sentence got lost here

Reply:
I found it again